onnxruntime/contrib_ops/cpu/transformers/generation_device_helper.cc

```diff
@@ -901,45 +896,15 @@ Status CreateWhisperEncoderInputs(
                        allocator->Info(),
                        encoder_input_features);
 
-  if (attn_mask_value != nullptr) {
-    const Tensor& attention_mask = attn_mask_value->Get<Tensor>();
-    Tensor::InitOrtValue(element_type, input_features_shape, const_cast<Tensor*>(&attention_mask)->MutableData<int32_t>(),
-                         allocator->Info(), encoder_attention_mask);
-  } else {
-    auto mask_type = DataTypeImpl::GetType<int32_t>();
-    Tensor::InitOrtValue(mask_type, input_features_shape, allocator, encoder_attention_mask);
-
-    // Set attention mask to be 0 for pad tokens, and 1 for all other tokens.
-    int32_t* mask_data = encoder_attention_mask.GetMutable<Tensor>()->MutableData<int32_t>();
-    const int32_t* word_id = original_encoder_input_features->Data<int32_t>();
-    int32_t* mask = mask_data;
-    for (int i = 0; i < batch_size; i++) {
-      int32_t abs_position = 0;
-      for (int j = 0; j < sequence_length; j++, word_id++, mask++) {
-        // T5Tokenizer might add one EOS pad token at the end.
-        // That EOS token shall have attention mask 1 even when EOS token is same as pad token.
-        // Here we only set attention mask to be 0 for left padding only, so as to be parity with huggingface.
-        if (*word_id == pad_token_id && abs_position == 0) {
-          *mask = 0;
-        } else {
-          *mask = 1;
-          abs_position++;
-        }
-      }
-    }
-  }
-
-  // decoder_input_ids is optional.
-  if (start_token_id >= 0) {
-    // Filled decoder_input_ids with start token ID
-    int64_t dims[] = {batch_size, 1};
-    TensorShape decoder_input_ids_shape(&dims[0], 2);
-    Tensor::InitOrtValue(element_type, decoder_input_ids_shape, allocator, decoder_input_ids);
-    int32_t* data = decoder_input_ids.GetMutable<Tensor>()->MutableData<int32_t>();
-    for (int i = 0; i < batch_size; i++, data++) {
-      *data = start_token_id;
-    }
-  }
+  // decoder_input_ids is of shape (batch_size, initial_sequence_length):
+  // Example: [[ decoder start token (i.e. start of transcript), language token, task token, timestamp token ]]
+  const TensorShape& original_decoder_input_ids_shape = original_decoder_input_ids->Shape();
+  ORT_ENFORCE(original_decoder_input_ids_shape.NumDimensions() == 2);
```

Comment:
I think this enforces `original_decoder_input_ids` as an input instead of making it optional. I believe behavior should default to original initialization (using start_token_id) if original_decoder_input_ids is not provided.